support/cas-server-support-oauth-core-api/src/main/java/org/apereo/cas/support/oauth/authenticator/OAuth20ClientIdClientSecretAuthenticator.java

```diff
@@ -121,6 +124,14 @@ protected boolean canAuthenticate(final WebContext context) {
             return false;
         }
 
+        if (grantType.isPresent()
+            && OAuth20Utils.isGrantType(grantType.get(), OAuth20GrantTypes.REFRESH_TOKEN)
+            && context.getRequestParameter(OAuth20Constants.CLIENT_ID).isPresent()
+            && !context.getRequestParameter(OAuth20Constants.CLIENT_SECRET).isPresent()) {
+            LOGGER.debug("Skipping Client credential authentication to use refresh_token authentication");
```

Comment:
Please correct the capitalization of the debug statement (i.e. Client), and possibly remove *refresh_token authentication*.

Reply:
i chose to keep `refresh token authentication` part. It seems more logical with the previous statement isn't it ?

```
LOGGER.debug("Skipping Client credential authentication to use password authentication");
```